### ISSUE TITLE
No support for named/multiple build contexts (new Dockerfile 1.4 feature)

### ISSUE THREAD

**Issue by wimax-grapl** [hadolint/hadolint#830]
- [x] This is a bug report
- [x] This is a feature request
- [x] I searched existing issues before opening this one

There's a new feature in Docker that lets you specify named contexts.
The contexts are piped in from a `docker buildx build` flag or a `docker-bake.hcl` specification.
https://www.docker.com/blog/dockerfiles-now-support-multiple-build-contexts/

### Expected behavior
No DL3022 when I COPY --from a named context

### Actual behavior
```
src/python/Dockerfile:54 DL3022 warning: `COPY --from` should reference a previously defined `FROM` alias
```

### Steps to reproduce the behavior
```
COPY --from=some-named-context some_file .
```

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**
2.10

```bash
(paste your output here)
```
5.1.4(1)-release

**Dockerfile** (if relevant)
https://github.com/grapl-security/grapl/pull/1686/files#diff-89d34b14f936765223e09dfbc3d2d7d37628da341845e2ad14fd766d7c784a52
also the docker-bake.hcl:
https://github.com/grapl-security/grapl/pull/1686/files#diff-870f6fe23fc034f008f5203ee1e628d7bfa65a49095d5a4688db498328449ed2

**Additional environment details (OS, stack version, etc.)**

**Comment by lorenzo** [comment]
I would recommend adding an in-line ignore for the rule. There is no way for Hadolint to know what flags you will use at build time.

It is also unfortunate that they reused the from flag for a new feature, a new flag would have allowed us to infer that there was nothing to validate in this case.

**Comment by wimax-grapl** [comment]
Yeah, I don't disagree and that's what I ended up doing.

**Comment by gabsoftware** [comment]
Only way it to add `# hadolint ignore=DL3022` before the COPY line.
